### PR TITLE
Add steam RT compilation workflow

### PR DESCRIPTION
**Pull request by feliwir**
- [x] test both i386 and amd64
- [x] test against latest RT and the beta RT version 

Currently does not seem to compile due to `freeimage` not being a system library and not being available for real

### CODE REVIEW

File: .github/workflows/steam.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      fail-fast: false
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # Shallow clones speed things up
+      - name: "Build inside SteamRT container"
+        run: >
+          ID=$(docker run -d -v ${{ github.workspace }}:/og -i ${{ matrix.steamrt }}:${{ matrix.sdkver }}) &&
+          docker exec -w /og/ -i $ID sh -c
```
Comment (Conan-Kudo) [inline .github/workflows/steam.yml:26]: You are making this too complicated.

Use `container: ${{ matrix.steamrt }}` instead.

**Review by Conan-Kudo — CHANGES_REQUESTED** [review]

File: .github/workflows/steam.yml (hunk 1 above)
Comment (feliwir) [reply .github/workflows/steam.yml:26]: This does not work - the container itself can't perform a checkout that's why we need to copy the sourcetree into the container

**Review by Conan-Kudo — APPROVED** [review]
LGTM

**Comment by Conan-Kudo** [comment]
@feliwir Is this ready for proper review? If so, can you mark it as such?

**Review by Conan-Kudo — APPROVED** [review]

**Pull request merged**